src/main/java/org/example/daos/JobRoleDao.java

```diff
@@ -12,30 +13,50 @@
 import java.util.List;
 
 public class JobRoleDao {
-    public List<JobRole> getJobRoles() throws SQLException {
-        List<JobRole> jobRoles = new ArrayList<>();
+
+    public List<JobRoleResponse> getJobRoles(
+            final int offset, final int limit) throws SQLException {
+        List<JobRoleResponse> jobRoles = new ArrayList<>();
         try (Connection connection = DatabaseConnector.getConnection()) {
             String query = "SELECT jobRoleId, roleName, location,"
                     + " Capability.capabilityName, Band.bandName, "
                     + "closingDate FROM `job-roles`"
                     + " JOIN Capability using(capabilityId)"
                     + " JOIN Band using(bandId)"
-                    + " WHERE statusId = 1;";
+                    + " WHERE statusId = 1"
+                    + " ORDER BY jobRoleId"
+                    + " LIMIT ? OFFSET ?;";
             PreparedStatement statement = connection.prepareStatement(query);
+            statement.setInt(1, limit);
+            statement.setInt(2, offset);
             ResultSet resultSet = statement.executeQuery();
             while (resultSet.next()) {
-                JobRole jobRole = new JobRole(
+                JobRoleResponse jobRoleResponse = new JobRoleResponse(
                         resultSet.getInt("jobRoleId"),
                         resultSet.getString("roleName"),
                         resultSet.getString("location"),
                         resultSet.getString("capabilityName"),
                         resultSet.getString("bandName"),
                         resultSet.getDate("closingDate"));
-                jobRoles.add(jobRole);
+                jobRoles.add(jobRoleResponse);
             }
         }
         return jobRoles;
     }
+    public int countTotalJobs() throws SQLException {
```

Comment:
Badly named method. We are getting the total count of open job roles... Please rename to something more representative of what the method is doing

Reply:
Resolved bad naming convention to getTotalOpenJobs